creator.cpp

```diff
@@ -99,55 +99,55 @@ Creator::Creator(Privileges &privilegesArg, QWidget *parent) :
-            this, SLOT(handleFinishedDownload(QByteArray)));
-    connect(manager, SIGNAL(partialData(QByteArray,qlonglong)),
-            this, SLOT(handlePartialData(QByteArray,qlonglong)));
-    connect(manager, SIGNAL(downloadError(QString)),
-            this, SLOT(handleDownloadError(QString)));
-
-    connect(ui->downloadButton, SIGNAL(clicked()),
-            this, SLOT(downloadButtonClicked()));
-    connect(ui->loadButton, SIGNAL(clicked()),
-            this, SLOT(getImageFileNameFromUser()));
-    connect(ui->writeFlashButton, SIGNAL(clicked()),
-            this, SLOT(writeFlashButtonClicked()));
-
-    connect(ui->imagesShowAll, SIGNAL(stateChanged(int)),
-            this, SLOT(projectImagesShowAllChanged(int)));
-    connect(ui->projectSelectBox, SIGNAL(currentIndexChanged(int)),
-            this, SLOT(setProjectImages()));
-    connect(ui->imageSelectBox, SIGNAL(currentTextChanged(QString)),
-            this, SLOT(projectImagesChanged(QString)));
-
-    connect(ui->removableDevicesComboBox, SIGNAL(currentIndexChanged(int)),
-            this, SLOT(savePreferredRemovableDevice(int)));
-
-    connect(ui->ejectUSB, SIGNAL(clicked()), this, SLOT(ejectUSB()));
-    connect(ui->loadUSB, SIGNAL(clicked()), this, SLOT(loadUSB()));
-    connect(ui->removeUSB, SIGNAL(clicked()), this, SLOT(removeUSB()));
-    connect(ui->helpButton, SIGNAL(clicked()), this, SLOT(showHelp()));
-    connect(ui->showAboutButton, SIGNAL(clicked()), this, SLOT(showAbout()));
-    connect(ui->closeAboutButton, SIGNAL(clicked()), this, SLOT(closeAbout()));
-    connect(ui->closeAppButton, SIGNAL(clicked()), this, SLOT(close()));
-
-    connect(ui->langButton,SIGNAL(clicked()), this, SLOT(languageChange()));
+    connect(ui->refreshRemovablesButton,&QToolButton::clicked,
+            this, &Creator::refreshRemovablesList);
+
+    connect(manager, &DownloadManager::downloadProgress,
+            this, &Creator::handleDownloadProgress);
+    connect(manager, &DownloadManager::downloadComplete,
+            this, &Creator::handleFinishedDownload);
+    connect(manager, &DownloadManager::partialData,
+            this, &Creator::handlePartialData);
+    connect(manager, &DownloadManager::downloadError,
+            this, &Creator::handleDownloadError);
+
+    connect(ui->downloadButton, &QPushButton::clicked,
+            this, &Creator::downloadButtonClicked);
+    connect(ui->loadButton, &QPushButton::clicked,
+            this, &Creator::getImageFileNameFromUser);
+    connect(ui->writeFlashButton, &QPushButton::clicked,
+            this, &Creator::writeFlashButtonClicked);
+
+    connect(ui->imagesShowAll, &QCheckBox::stateChanged,
+            this, &Creator::projectImagesShowAllChanged);
+    connect(ui->projectSelectBox, QOverload<int>::of(&QComboBox::currentIndexChanged),
+            this, &Creator::setProjectImages);
+    connect(ui->imageSelectBox, QOverload<const QString&>::of(&QComboBox::currentTextChanged),
```

Comment:
why overload? Only a single signal with such name exists.

Reply:
The overload on `currentTextChanged()` is unnecessary, I'll remove this one. I did some testing when this was still `currentIndexChanged()`, didn't think to recheck it after that.

The overload on `currentIndexChanged()` is required with Qt 5. 
<https://doc.qt.io/qt-5/qcombobox.html#currentIndexChanged>
Reading the other comment from @chewitt, if we only target Qt 6, the `currentIndexChanged()` overloading, can be removed as well.

Reply:
I was talking specifically about `currentTextChanged` (as commenting not on a range of lines, probably not so easy to spot)